mitodl/mit-open
# For xpro, conditionally ingest topics for each resource depending on formatting

Blocked due to inconsistent formatting of prolearn topics in production.

#### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5551

### Description (What does it do?)
Conditionally assigns topics based on the formatting of the xpro API topics for each course/program:
  - If any topics in the list do not have “:” -> just ingest those, ignore the others.
  - If all topics in the list have “:” -> transform them from prolearn to mit-learn format


### How can this be tested?
- Set the following backend env variables:
  ```
  XPRO_CATALOG_API_URL=https://xpro.mit.edu/api/programs/
  XPRO_COURSES_API_URL=https://xpro.mit.edu/api/courses/
  ``
- Run `./manage.py backpopulate_xpro_data`, it should complete without errors.
- Go to https://learn.mit.edu/search?offered_by=xpro and check that most xpro resources have assigned topics.
